AssertionError: Torch not compiled with CUDA enabled

### 🐛 Describe the bug

Cases:
op_ut,third_party.torch-xpu-ops.test.xpu.test_distributions_xpu.TestDistributionsXPU,test_torch_binomial_dtype_errors_xpu

Command
pytest -v test_distributions_xpu.py -k test_torch_binomial_dtype_errors_xpu

```
2026-03-30T15:23:36.4301353Z =================================== FAILURES ===================================
2026-03-30T15:23:36.4301724Z __________ TestDistributionsXPU.test_torch_binomial_dtype_errors_xpu ___________
2026-03-30T15:23:36.4302405Z [gw4] linux -- Python 3.10.20 /tmp/xpu-tool/Python/3.10.20/x64/bin/python
2026-03-30T15:23:36.4302719Z Traceback (most recent call last):
2026-03-30T15:23:36.4303372Z   File "/__w/torch-xpu-ops/torch-xpu-ops/pytorch/third_party/torch-xpu-ops/test/xpu/../../../../test/distributions/test_distributions.py", line 1830, in test_torch_binomial_dtype_errors
2026-03-30T15:23:36.4304021Z     total_count = torch.tensor([10, 10], dtype=count_dtype, device=device)
2026-03-30T15:23:36.4304525Z   File "/tmp/xpu-tool/Python/3.10.20/x64/lib/python3.10/site-packages/torch/cuda/__init__.py", line 475, in _lazy_init
2026-03-30T15:23:36.4305015Z     raise AssertionError("Torch not compiled with CUDA enabled")
2026-03-30T15:23:36.4305323Z AssertionError: Torch not compiled with CUDA enabled
2026-03-30T15:23:36.4305478Z 
2026-03-30T15:23:36.4305610Z To execute this test, run the following from the base repo dir:
2026-03-30T15:23:36.4306074Z     PYTORCH_TEST_WITH_SLOW=1 python ../../test/distributions/test_distributions.py TestDistributionsXPU.test_torch_binomial_dtype_errors_xpu
```

### Versions

https://github.com/intel/torch-xpu-ops/pull/3205

--- a/test/xpu/test_distributions_xpu.py
+++ b/test/xpu/test_distributions_xpu.py
@@ -155,6 +155,7 @@ def _test_poisson_gpu_sample(self):
             failure_rate=1e-3,
         )
 
+
 def _test_torch_binomial_dtype_errors(self):
     dtypes = [torch.int, torch.long, torch.short]
     devices = ["cpu", "xpu"]
@@ -179,6 +180,7 @@ def _test_torch_binomial_dtype_errors(self):
                 "binomial only supports floating-point dtypes for prob.*",
             ):
                 torch.binomial(total_count, total_prob)
+
 
 TestDistributions.test_beta_underflow_gpu = _test_beta_underflow_gpu
 TestDistributions.test_zero_excluded_binomial = _test_zero_excluded_binomial

--- a/test/xpu/test_distributions_xpu.py
+++ b/test/xpu/test_distributions_xpu.py
@@ -155,12 +155,37 @@ def _test_poisson_gpu_sample(self):
             failure_rate=1e-3,
         )
 
+def _test_torch_binomial_dtype_errors(self):
+    dtypes = [torch.int, torch.long, torch.short]
+    devices = ["cpu", "xpu"]
+
+    for device in devices:
+        for count_dtype in dtypes:
+            total_count = torch.tensor([10, 10], dtype=count_dtype, device=device)
+            total_prob = torch.tensor([0.5, 0.5], dtype=torch.float, device=device)
+
+            with self.assertRaisesRegex(
+                ValueError,
+                "binomial only supports floating-point dtypes for count.*",
+            ):
+                torch.binomial(total_count, total_prob)
+
+        for prob_dtype in dtypes:
+            total_count = torch.tensor([10, 10], dtype=torch.float, device=device)
+            total_prob = torch.tensor([0.5, 0.5], dtype=prob_dtype, device=device)
+
+            with self.assertRaisesRegex(
+                ValueError,
+                "binomial only supports floating-point dtypes for prob.*",
+            ):
+                torch.binomial(total_count, total_prob)
 
 TestDistributions.test_beta_underflow_gpu = _test_beta_underflow_gpu
 TestDistributions.test_zero_excluded_binomial = _test_zero_excluded_binomial
 TestDistributions.test_gamma_gpu_sample = _test_gamma_gpu_sample
 TestDistributions.test_gamma_gpu_shape = _test_gamma_gpu_shape
 TestDistributions.test_poisson_gpu_sample = _test_poisson_gpu_sample
+TestDistributions.test_torch_binomial_dtype_errors = _test_torch_binomial_dtype_errors
 instantiate_device_type_tests(
     TestDistributions, globals(), only_for="xpu", allow_xpu=True
 )

